From 90060fa6605446bef7078867423b691e4effa575 Mon Sep 17 00:00:00 2001
From: Tom Gundersen <teg@jklm.no>
Date: Fri, 13 Sep 2013 14:46:18 +0200
Subject: [PATCH] swap: fix reverse dependencies

Make sure swap.target correctly requires/wants the swap units.

This fixes https://bugs.freedesktop.org/show_bug.cgi?id=69291.

Reported-by: Hussam Al-Tayeb
---
 src/core/swap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/swap.c b/src/core/swap.c
index 57d15eb..3950860 100644
--- a/src/core/swap.c
+++ b/src/core/swap.c
@@ -220,7 +220,7 @@ static int swap_add_default_dependencies(Swap *s) {
         }
 
         if (!noauto) {
-                r = unit_add_two_dependencies_by_name(UNIT(s), UNIT_BEFORE, (nofail ? UNIT_WANTED_BY : UNIT_REQUIRED_BY),
+                r = unit_add_two_dependencies_by_name_inverse(UNIT(s), UNIT_AFTER, (nofail ? UNIT_WANTS : UNIT_REQUIRES),
                                                       SPECIAL_SWAP_TARGET, NULL, true);
                 if (r < 0)
                         return r;
-- 
1.8.4

